src/NHSD.BuyingCatalogue.Organisations.Api/Repositories/OdsRepository.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Flurl;
+using Flurl.Http;
+using NHSD.BuyingCatalogue.Organisations.Api.Models;
+
+namespace NHSD.BuyingCatalogue.Organisations.Api.Repositories
+{
+    internal sealed class OdsRepository : IOdsRepository
+    {
+        private readonly string _buyerOrganisationRoleId;
+        private readonly string _odsApiBaseUrl;
+
+        public OdsRepository(string odsApiBaseUrl)
+        {
+            _buyerOrganisationRoleId = "RO98";
+            _odsApiBaseUrl = odsApiBaseUrl;
+        }
+
+        public async Task<OdsOrganisation> GetBuyerOrganisationByOdsCodeAsync(string odsCode)
+        {
+
+            var response = await _odsApiBaseUrl
+                .AppendPathSegment("organisations")
+                .AppendPathSegment(odsCode)
+                .AllowHttpStatus("3xx,4xx")
+                .GetJsonAsync<OdsResponse>();
+
+            var odsOrganisation = response.Organisation;
+
+            return odsOrganisation is null ? null : new OdsOrganisation
+            {
+                OrganisationName = odsOrganisation.Name,
+                OdsCode = odsCode,
+                PrimaryRoleId = _buyerOrganisationRoleId,
+                Address = OdsResponseAddressToAddress(odsOrganisation.GeoLoc.Location),
+                IsActive = IsActive(odsOrganisation),
+                IsBuyerOrganisation = IsBuyerOrganisation(odsOrganisation)
+            };
+        }
+
+        private static bool IsActive(OdsResponseOrganisation organisation)
+        {
+            return organisation.Status.Equals("Active", StringComparison.OrdinalIgnoreCase);
+        }
+
+        private bool IsBuyerOrganisation(OdsResponseOrganisation organisation)
+        {
+            return organisation.Roles.Role.Any(role =>
+                 role.id.Equals(_buyerOrganisationRoleId, StringComparison.OrdinalIgnoreCase));
```

Comment:
I think this might look better if the entire lambda was moved to the new line, like this:

```c#
return organisation.Roles.Role.Any(
    role => role.id.Equals(_buyerOrganisationRoleId, StringComparison.OrdinalIgnoreCase));
```
However, it's up to you.

Reply:
Due to change in "the business logic" (buyer organisation is identified as having a primary role id of either RO98 or RO213) I'll be refactoring this anyway 😄